### ISSUE TITLE
core.latex.renderer module does not exist

### ISSUE THREAD

**Issue by mlysle** [nvim-neorg/neorg#1455]
### Prerequisites

- [X] I am using the latest stable release of Neovim
- [X] I am using the latest version of the plugin

### Neovim Version

NVIM v0.10.0 Build type: RelWithDebInfo LuaJIT 2.1.1707061634

### Neorg setup

```lua
		neorg.setup {
			load = {
				["core.defaults"] = {},
				["core.concealer"] = {},
				["core.esupports.metagen"] = {
					config = {
						author = 'Maxwell Lysle',
						type = 'auto',
					}
				},
				["core.keybinds"] = {
					config = {
						hook = function(keybinds)
							--keybinds.map_event("norg", "n", "<C-s>", "core.integrations.telescope.find_linkable")
							--keybinds.map_event("norg", "i", "<C-l>", "core.integrations.telescope.insert_link")
							keybinds.remap_key("norg", "i", "<M-CR>", "<S-CR>")
							keybinds.remap_key("norg", "n", "<Tab>", "core.integrations.treesitter.next.heading")
							keybinds.remap_key("norg", "n", "<S-Tab>", "core.integrations.treesitter.previous.heading")
						end,
					},
				},
				["core.completion"] = {
					config = {
						engine = 'nvim-cmp',
					}
				},
				["core.journal"] = {
					config = {
						workspace = 'notes',
						strategy = 'flat',
					}
				},
				-- ["core.presenter"] = {},
				-- ["core.integrations.telescope"] = {},
				["core.dirman"] = {
					config = {
						workspaces = {
							notes = "~/documents/neorg/notes",
							gtd = "$~/documents/neorg/gtd",
						},
						default_workspace = "notes",
					}
				},
				["core.integrations.image"] = {},
				["core.latex.renderer"] = {}
			}
		}
```

### Actual behavior

Loading neovim results in the following error:
```
[neorg] (12:12:15)                                                                   
.../.local/share/nvim/lazy/neorg/lua/neorg/core/modules.lua:449                      
Unable to load module 'core.latex.renderer'. The module probably does not exist! Stac
ktrace:  /usr/share/nvim/runtime/lua/vim/re.lua:150: attempt to perform arithmetic on
 local 'item' (a userdata value)                                                     
Press ENTER or type command to continue                                              
[neorg] (12:12:15)                                                                   
...e/mlysle/.local/share/nvim/lazy/neorg/lua/neorg/init.lua:157                      
Recovering from error..
```
No other modules fail to load.

### Expected behavior

There shouldn't be any trouble loading this module. I have confirmed that `~/.local/share/lazy/neorg/lua/neorg/modules/core/latex` exists.

### Steps to reproduce

- Install dependencies
- Configure module
- Load neovim

### Other information

I am using Lazy.nvim with luarocks.nvim. I have tried running `:Lazy build neorg', but it didn't fix the problem.

### Help

None

**Comment by benlubas** [comment]
> I have confirmed that ~/.local/share/lazy/neorg/lua/neorg/modules/core/latex exists.

That's the wrong path for the module. Do you mind sharing the file contents of that file?

This is most likely just a case of not running the most up to date version of neorg

**Comment by mlysle** [comment]
> > I have confirmed that ~/.local/share/lazy/neorg/lua/neorg/modules/core/latex exists.
> 
> That's the wrong path for the module (I guess unless you just left out the `renderer` folder for some reason?). Do you mind sharing the file contents of that file?
> 
> This is most likely just a case of not running the most up to date version of neorg

I typed the path wrong (I didn't copy/paste). This is the actual path: `.local/share/nvim/lazy/neorg/lua/neorg/core/latex`.

And I am running the latest version of neorg. Lazy says I am at commit b17a33e which is the same as the latest commit in the main branch.

**Comment by max397574** [comment]
as you can see there was an error with `vim.re` ig it's because of that
quite sure we had similar issues in the past

**Comment by max397574** [comment]
same error as here https://github.com/nvim-neorg/neorg/issues/1106#issue-1932059716

**Comment by benlubas** [comment]
all we do with vim.re is use it as a proxy for nvim version 10+ by checking for its existence. maybe an nvim bug that no longer exists? are you sure you're on a stable version of 0.10 @mlysle?

**Comment by mlysle** [comment]
> all we do with vim.re is use it as a proxy for nvim version 10+ by checking for its existence. maybe an nvim bug that no longer exists? are you sure you're on a stable version of 0.10 @mlysle?

I'm on whatever version of neovim ships with Fedora 40. There is an unofficial repo with nightly nvim builds. I'll look into it tomorrow and report my results.

**Comment by benlubas** [comment]
@mlysle you should use the 0.10 stable release, not the 0.10 `RelWithDebInfo`, and not 11 nightly.

For example, this is the output of `nvim --version` when I run it:
```
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1693350652
Run "nvim -V1 -v" for more info
```

**Comment by mlysle** [comment]
It looks like Fedora just ships neovim with debuginfo enabled. I'll have to compile neovim myself I guess.

**Comment by mlysle** [comment]
fixed by switching to 0.10 stable
